File: 001_binary_exponentiation/implementations/go/main.go
package main

import "fmt"

func binExp(a int, b int) int {
	var result int = 1
	for b > 0 {
		if b%2 != 0 {
			result *= a
		}
		a *= a
		b /= 2
	}
	return result
}

func main() {
	fmt.Println(binExp(3, 22))
}

File: 001_binary_exponentiation/implementations/c/main.c
#include <stdio.h>

long long binExp(int a, int b)
{
    long long result = 1;
    while (b > 0)
    {
        if (b % 2 != 0)
            result *= a;
        a *= a;
        b /= 2;
    }
    return result;
}

int main()
{
    printf("%ld\n", binExp(3, 9));
    return 0;
}